packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -493,15 +494,21 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		localId: string,
 		blob: ArrayBufferLike,
 	): Promise<ICreateBlobResponse | void> {
-		return PerformanceEvent.timedExecAsync(
-			this.mc.logger,
-			{ eventName: "createBlob" },
+		return runWithRetry(
 			async () => this.getStorage().createBlob(blob),
-			{ end: true, cancel: this.runtime.connected ? "error" : "generic" },
-		).then(
-			(response) => this.onUploadResolve(localId, response),
-			async (err) => this.onUploadReject(localId, err),
-		);
+			"createBlob",
+			this.mc.logger,
+			{
+				cancel: this.pendingBlobs.get(localId)?.abortSignal,
+			},
+		)
+			.then((response) => this.onUploadResolve(localId, response))
+			.catch((err) => {
+				this.pendingBlobs.get(localId)?.handleP.reject(err);
+				if (!this.pendingBlobs.get(localId)?.opsent) {
```

Comment:
if the op has been sent and so we don't delete the pending blob, how does the upload get restarted?

Reply:
The only way the op has been sent at this point is because it's being stashed. We want to avoid messing with pendingBlobs array during that process.